samples/reducer.js

```diff
@@ -54,9 +68,47 @@ function createReducer (config) {
     }),
     [loadSampleEnd]: (state, action) => ({
       ...state, isLoading: false
+    }),
+    [createSample]: (state, action) => {
+      const file = action.payload
+      assert(file && file.path, 'Cannot createSample without file && file.path')
+
+      return loop({
+        ...state, isCreating: true
+      }, Effects.batch([
+        Effects.promise(runCreateSample, file),
+        Effects.constant(createSampleEnd())
+      ]))
+    },
+    [createSampleSuccess]: (state, action) => {
+      const { id, fileName } = action.payload
+      const meta = {
+        id,
+        title: fileName
+      }
+
+      return loop({
+        ...state,
+        records: {
+          ...state.records,
+          [action.payload.id]: action.payload
+        }
+      }, Effects.batch([
+        // TODO: here is a case where ordering might be nice.
```

Comment:
thoughts? should i always be creating my effects in a way that doesnt need a specific order?